### PR TITLE
Parse block type with leb i32 encoding and correct dst dynamic offset for opcode br

### PR DESCRIPTION
**Pull request by wenyongh**
Fix several issues in wasm loader:
- Parse a block's type index with leb int32 instead leb uint32
- Correct dst dynamic offset of loop block arguments for opcode br
  when copying the stack operands to the arguments of loop block
- Free each frame_csp's param_frame_offsets when destroy loader ctx
- Fix compilation error in wasm_mini_loader.c
- Add test cases of failed issues

This PR fixes issue #3467 and #3468.

### CODE REVIEW

File: core/iwasm/interpreter/wasm_loader.c [hunk 1]
```diff
@@ -8116,6 +8134,8 @@ wasm_loader_ctx_destroy(WASMLoaderContext *ctx)
         if (ctx->frame_csp_bottom) {
 #if WASM_ENABLE_FAST_INTERP != 0
             free_all_label_patch_lists(ctx->frame_csp_bottom, ctx->csp_num);
+            free_all_label_param_frame_offsets(ctx->frame_csp_bottom,
```
Comment (xujuntwt95329) [inline core/iwasm/interpreter/wasm_loader.c:8137]: wasm_loader_pop_frame_csp will also free the param_frame_offsets, should we set it to NULL in wasm_loader_pop_frame_csp?

Comment (wenyongh) [reply core/iwasm/interpreter/wasm_loader.c:8137]: The wasm_loader_push_frame_csp will memset the frame csp with 0, seems no need to set param_frame_offsets to NULL again:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/67638e24f40a537b194a233a2d388f06db6d9546/core/iwasm/interpreter/wasm_loader.c#L8672-L8678

Comment (xujuntwt95329) [reply core/iwasm/interpreter/wasm_loader.c:8137]: Got it, thanks!

**Review by xujuntwt95329 — COMMENTED** [review]
LGTM

**Pull request merged**